css/css-text/i18n/ja/css-text-line-break-ja-cj-loose.html

```diff
@@ -0,0 +1,108 @@
+<!DOCTYPE html>
+<html  lang="en" >
+<head>
+<meta charset="utf-8"/>
+<title>Line-break:loose, Conditional Japanese Starter (CJ) (ja)</title>
+<link rel="author" title="Richard Ishida" href="mailto:ishida@w3.org">
+<link rel="help" href="https://drafts.csswg.org/css-text-3/#line-break-property">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<meta name="assert" content="The browser allows a conditional Japanese starter at the beginning of a line.">
+<style type="text/css">
+@font-face {
+    font-family: 'mplus-1p-regular';
+    src: url('/fonts/CSSTest/mplus-1p-regular.woff') format('woff');
+    }
+#wrapper { position: relative; }
+.test { color: red; }
+.test, .ref { font-size: 30px; font-family: mplus-1p-regular, sans-serif; width: 185px; padding: 0; border: 1px solid orange; line-height: 1em; }
+</style>
+<style>
+.test { line-break: loose; }
+</style>
+<script>
+var charlist = `3041  HIRAGANA LETTER SMALL A
```

Comment:
I think the current string format is easier to maintain when upgrading Unicode versions, since most Unicode data is provided in text files. I agree your suggestion is easier for JavaScript to read, but then someone must translate the source text data to JSON. Isn't it better for computer to handle such machinery translations?

Reply:
Of course, if that's copy&pasted from the Unicode spec, then it makes all the sense to have it like a string.
I didn't know it, that's why I was suggesting something different. Thanks for the explanation.

Reply:
Can we have some comment saying exactly where the charlist comes from (so others can update it, etc.)?

Reply:
The actual lists comes from Unicode, but what part of unicode to copy and paste is described in the spec.

> breaks before Japanese small kana or the Katakana-Hiragana prolonged sound mark: i.e. characters with the Unicode Line Break property CJ. (See LineBreak.txt in [UNICODE].) 

we could copy that sentence, link included, into the test, but going through the spec, to which the text does link, may be better to make sure you don't miss something if the spec is changed somehow.

Reply:
I used this for the main data source: https://www.unicode.org/Public/UNIDATA/extracted/DerivedLineBreak.txt, expanding the ranges and reformatting to get the list in the tests. For more information about expected line break behavior and line break classes, see [Unicode Standard Annex #14 Line Breaking Properties](http://www.unicode.org/reports/tr14/).